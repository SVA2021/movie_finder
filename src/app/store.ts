import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
