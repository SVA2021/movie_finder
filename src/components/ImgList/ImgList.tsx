import {Box, useTheme} from '@mui/material';
import {FC, memo, useEffect, useRef, useState} from 'react';
import {useParams} from 'react-router-dom';
import {Swiper, SwiperSlide} from 'swiper/react';
import 'swiper/swiper.min.css'; // core Swiper
import {EmptyDataBox, ModalBox} from '..';
import {useAppDispatch, useAppSelector} from '../../app/hooks';
import {selectDetails} from '../../features/movieFinder/movieFinderSlice';
import {getMovieImagesAsync} from '../../features/movieFinder/movieFinderThunks';

export const ImgList: FC = memo(() => {

  const dispatch = useAppDispatch();
  const DetailsPage = useAppSelector(selectDetails);
  const Images = DetailsPage.images?.items ?? [];
  const theme = useTheme();
  const swiperRef = useRef<any>();
  const params = useParams();
  const id = Number(params.id)

  const [fullImage, setFullImage] = useState<string | null>(null);

  useEffect(() => {
    dispatch(getMovieImagesAsync(id))
  }, [id]);

  return (
    <Box >
      {Images.length > 0
        ? <Swiper
          onSwiper={(swiper) => swiperRef.current = swiper}
          slidesPerView={1}
          spaceBetween={8}
          breakpoints={
            {
              [theme.breakpoints.values.md]: {
                slidesPerView: 3,
                spaceBetween: 16,
              },
              [theme.breakpoints.values.lg]: {
                slidesPerView: 5,
                spaceBetween: 24,
              },
            }
          }
        >
          {
            Images.map((item, index) =>
              <SwiperSlide key={index} >
                <Box
                  sx={{
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'center',
                    cursor: 'pointer',
                  }}
                  onClick={() => setFullImage(item.imageUrl)}
                >
                  <img
                    height={'200px'}
                    src={item.previewUrl}
                    alt={item.previewUrl}
                    title={'открыть полную версию'}
                  />
                </Box>
              </SwiperSlide>
            )
          }
        </Swiper>
        : <EmptyDataBox />
      }
      {fullImage &&
        <ModalBox closeHandler={() => setFullImage(null)}>
          <img
            src={fullImage}
            alt="full version"
            width={'auto'}
            height={'100%'}
            onClick={() => setFullImage(null)}
          />
        </ModalBox>
      }
    </Box>
  );
});