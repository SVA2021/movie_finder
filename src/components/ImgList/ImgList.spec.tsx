import {render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {store} from "../../app/store";
import {ImgList} from "./ImgList";

describe('render ImgList', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <ImgList />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('img-list')).toMatchSnapshot();
  });

})