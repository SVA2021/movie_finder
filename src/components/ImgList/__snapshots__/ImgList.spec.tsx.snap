// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ImgList snapshot test 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="img-list"
>
  <p
    class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
    data-testid="empty-data-box"
  >
    Нет данных
  </p>
</div>
`;
